docs/stake-pool-course/handbook/grafana-dashboard-tutorial.md

```diff
@@ -0,0 +1,492 @@
+:::
+
+
+repeat Step 2 for all your RN
+
+
+
+## 3. Install Grafana on Relay Node
+
+
+```shell
+$ wget -q -O - https://packages.grafana.com/gpg.key | sudo apt-key add -
+
+$ echo "deb https://packages.grafana.com/oss/deb stable main" > grafana.list
+$ sudo mv grafana.list /etc/apt/sources.list.d/grafana.list
+
+$ sudo apt-get update && sudo apt-get install -y grafana
+```
+Enable services so they start automatically
+```shell
+$ sudo systemctl enable grafana-server.service
+$ sudo systemctl enable prometheus.service
+$ sudo systemctl enable prometheus-node-exporter.service
+```
+Update prometheus.yml located in /etc/prometheus/prometheus.yml
+
+Change the *ip address* in the following command
+```shell
+$ cat > prometheus.yml << EOF
+global:
+  scrape_interval:     15s # By default, scrape targets every 15 seconds.
+
+  # Attach these labels to any time series or alerts when communicating with
+  # external systems (federation, remote storage, Alertmanager).
+  external_labels:
+    monitor: 'codelab-monitor'
+
+# A scrape configuration containing exactly one endpoint to scrape:
+# Here it's Prometheus itself.
+scrape_configs:
+  # The job name is added as a label job=<job_name> to any timeseries scraped from this config.
+  - job_name: 'prometheus'
+
+    static_configs:
+      - targets: ['localhost:9100']
+
+        labels:
+          alias: 'relaynode1'
+          type:  'cardano-node'
+
+      - targets: ['<relay node 2 public ip address>:9100']
+
+        labels:
+          alias: 'relaynode2'
+          type:  'cardano-node'
+      - targets: ['<block producer public ip address>:9100']
+
+        labels:
+          alias: 'block-producer-node'
+          type:  'cardano-node'
+     - targets: ['localhost:12798']
+        labels:
+          alias: 'relaynode1'
+          type:  'cardano-node'
+
+     - targets: ['<relay node 2 public ip address>:12798']
+
+        labels:
+          alias: 'relaynode2'
+          type:  'cardano-node'
+
+     - targets: ['<block producer public ip address>:12798']
+        labels:
+          alias: 'block-producer-node'
+          type:  'cardano-node'
+
+EOF
+```
+if you have more than two RNs, add all your Relays as new "targets" in the config above
+```shell
+$ sudo mv prometheus.yml /etc/prometheus/prometheus.yml
+```
+restart the services
+```shell
+$ sudo systemctl restart grafana-server.service
+$ sudo systemctl restart prometheus.service
+$ sudo systemctl restart prometheus-node-exporter.service
+```
+Verify that the services are running properly
+```shell
+$ sudo systemctl status grafana-server.service prometheus.service prometheus-node-exporter.service
+```
+On RN open ports 3000 for Grafana
+```shell
+$ sudo ufw allow from <your local IP address> to any port 3000
+```
+
+
+## 4. Setting up Grafana Dashboard
+
+
+
+On RN, open http://localhost:3000 or http://*your relaynode ip address*:3000 in your local browser.
```

Comment:
"relay node" - two words for consistency please

Reply:
sure